doc: Rephrase some text referring to `DArray`

Socob opened the pull request:
Some parts of `distributed-computing.md` were somewhat unclear/confusing after `DArray` was moved to `DistributedArrays.jl`.

KristofferC commented:
Could you revert all the quote changes and only leave the "meaty" changes relating to `DArray`?

KristofferC commented:
It's generally better to split out mechanical changes like that, yes. But anyway, I think we just use the standard ASCII elsewhere?

Socob commented:
I had just checked `git grep '“' doc/` before making the changes, and did see that they’re used elsewhere in the docs (didn’t look for `"` because that would just show a bunch of `String` literals in code snippets).

Again, not sure if you want it in this PR or a separate one?

KristofferC commented:
Separate one, but looking through some of the markdown files I think we should do the opposite direction (“ -> ").

Socob commented:
Hm, well for typographical reasons I would really disagree with that, and rather go through all the Markdown files and consistently use *correct* quotation marks. Not sure if that’d be a welcome change though?

Socob committed:
doc: Rephrase some text referring to DArray

Some parts of distributed-computing.md were somewhat
unclear/confusing after DArray was moved to DistributedArrays.jl.

Socob commented:
I removed the quote changes now.
